docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,555 @@
+
+> :question: Is the current merge behavior what we need for routing policy? What surprises are there? See the next sections
+
+##### Service owner routes
+
+There's a discussion in the Gateway API repository about what different types of
+routes exist and how they should interact. Consider:
+
+- the owner of a service `svc_a`
+may want to configure a routing policy for their service that applies to requests from all services.
+- another service owner may want to apply routing to requests from their service
+to `svc_a`.
+
+How should these routes be merged? One invariant might be that the consumer of a
+service shouldn't be able to override the routes from the service owner.
+
+The route that applies to all requests:
+
+```yaml
+metadata:
+  name: owner
+spec:
+ targetRef:
+   kind: Mesh
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    default:
+      rules:
+      - matches:
+          path:
+            prefix: /
+        filters:
+        - responseHeader:
+            add:
+            - name: X-Some-Header
+              value: something
+        backendRefs:
+        - weight: 90
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v1
+        - weight: 10
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v2
+```
+
+The route that applies to requests from a specific service:
+
+```yaml
+metadata:
+  name: consumer
+spec:
+ targetRef:
+   kind: MeshService
+   name: svc_b
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    default:
+      rules:
+      - matches:
+          path:
+            prefix: /
+        filters:
+        - requestHeader:
+            add:
+            - name: X-Client
+              value:
+        backendRefs:
+        - weight: 100
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v1
+```
+
+when these resources are merged the latter will have higher priority and
+override the less specific resource:
+
+```yaml
+metadata:
+  name: consumer
+spec:
+ targetRef:
+   kind: MeshService
+   name: svc_b
+ to:
+  - targetRef:
+      kind: MeshService
+      name: svc_a
+    default:
+      rules:
+      - matches:
+          path:
+            prefix: /
+        filters:
+        - requestHeader:
+            add:
+            - name: X-Client
+              value:
+        backendRefs:
+        - weight: 100
+          kind: MeshServiceSubset
+          name: svc_a
+          tags:
+            version: v1
+```
+
+But the service owner probably expects to somehow
+be able to "enforce" routes on consumers that can't be overriden.
+
+Perhaps we finally need [`override`](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#hierarchy)?
+
+##### Merging
+
+Is there a use case for being able to write a policy that combines with the above `owner` policy such that the final resource would have both `filters`?
```

Comment:
One scenario that I came up with was mesh operator canary testing some infra feature. Image it is turned on by a header so we would like mesh operators to add this header to requests in "X" zone.

Reply:
for the next MADR